node/src/actors/chain_manager/mod.rs

```diff
@@ -903,7 +992,120 @@ impl ChainManager {
         self.magic
     }
 
+    /// Build and vote candidate superblock process which uses futures
+    #[must_use]
+    pub fn build_and_vote_candidate_superblock(
+        &mut self,
+        ctx: &mut Context<Self>,
+        superblock_epoch: u32,
+    ) -> ResponseActFuture<Self, (), ()> {
+        let fut = self.construct_superblock(ctx, superblock_epoch).and_then(
+            move |superblock, act, _ctx| {
+                let superblock_hash = superblock.hash();
+                log::debug!(
+                    "Local SUPERBLOCK #{} {}: {:?}",
+                    superblock.index,
+                    superblock_hash,
+                    superblock
+                );
+
+                let mut superblock_vote =
+                    SuperBlockVote::new_unsigned(superblock_hash, superblock.index);
+                let bn256_message = superblock_vote.bn256_signature_message();
+
+                signature_mngr::bn256_sign(bn256_message)
+                    .map_err(|e| {
+                        log::error!("Failed to sign superblock with bn256 key: {}", e);
+                    })
+                    .and_then(move |bn256_keyed_signature| {
+                        // Actually, we don't need to include the BN256 public key because
```

Comment:
```suggestion
                        // TODO: Actually, we don't need to include the BN256 public key because
```

Reply:
The comment was confusing. Actually, it wasn't TODO but an explanation. :rofl: